modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js

```diff
@@ -419,27 +436,26 @@ export default class NestedTabs extends Base {
 	}
 
 	updateIndexValues() {
-		const { $tabContents, $tabTitles } = this.getDefaultElements(),
+		const { $widgetContainer, $tabContents, $tabTitles } = this.getDefaultElements(),
 			settings = this.getSettings(),
-			itemIdBase = $tabTitles[ 0 ].getAttribute( 'id' ).slice( 0, -1 ),
-			containerIdBase = $tabContents[ 0 ].getAttribute( 'id' ).slice( 0, -1 );
+			widgetNumber = $widgetContainer.data( 'widgetNumber' );
 
 		$tabTitles.each( ( index, element ) => {
 			const newIndex = index + 1,
-				updatedTabID = itemIdBase + newIndex,
-				updatedContainerID = containerIdBase + newIndex;
+				updatedTabID = `e-n-tab-title-${ widgetNumber }${ newIndex }`,
+				updatedContainerID = `e-n-tab-content-${ widgetNumber }${ newIndex }`;
 
 			element.setAttribute( 'id', updatedTabID );
 			element.setAttribute( 'style', `--n-tabs-title-order: ${ newIndex }` );
 			element.setAttribute( 'data-tab-index', newIndex );
+			element.setAttribute( 'aria-controls', updatedContainerID );
```

Comment:
The `aria-controls` attribute should be set on the main element, and not on the title text selector.